File: compiler/python/state-buffer-estimate/arch/psumbuffer.py
from pearray import PeArray


##########################################################
class PsumBuffer(object):

    #-----------------------------------------------------------------
    def __init__(self, peArray, numberBanks, numberBankEntries):
        self.__NumberColumns        = peArray.gNumberColumns()
        self.__NumberBanks          = numberBanks
        self.__NumberBankEntries    = numberBankEntries
        self.__BankEntrySizeInBytes = 64 ## ???

    #-----------------------------------------------------------------
    def gNumberBanks(self):
        return self.__NumberBanks

    #-----------------------------------------------------------------
    def gNumberBankEntries(self):
        return self.__NumberBankEntries

    #-----------------------------------------------------------------
    def gBankEntrySizeInBytes(self):
        return self.__BankEntrySizeInBytes

    #-----------------------------------------------------------------
    def gNumberColumns(self):
        return self.__NumberColumns

File: compiler/kcc/arch/src/psumbuffer.cpp

#include "pearray.hpp"
#include "psumbuffer.hpp"

namespace kcc {
namespace arch {

//--------------------------------------------------------
PsumBuffer::PsumBuffer(const PeArray* peArray, int numberBanks, int numberBankEntries)
    : m_NumberColumns(peArray->gNumberColumns())
    , m_NumberBanks(numberBanks)
    , m_NumberBankEntries(numberBankEntries)
    , m_BankEntrySizeInBytes(64) // ???
{ }

}}

